### ISSUE TITLE
List required dependencies in README or `bin/help.deps`

### ISSUE THREAD

**Issue by jthegedus** [asdf-vm/asdf-nodejs#310]
asdf plugins should clearly list their dependencies required for the plugin to operate as expected. The core repo has had issues opened on this matter, most recently https://github.com/asdf-vm/asdf/issues/1176 Because of this, the [plugin template](https://github.com/asdf-vm/asdf-plugin-template) has a section in the README specifically to encourage listing of dependencies. The core also added the [call-back script `bin/help.deps`](https://asdf-vm.com/plugins/create.html#optional-scripts):

>`bin/help.deps` - This script should output the list of dependencies tailored to the operating system. One dependency per line.

It would be great if the `asdf-vm/asdf-*` plugins utilised these conventions to improve the UX of newcomers.

To illustrate how this can be an issue, the asdf docs getting-started guide has a section about [installing a plugin and it's dependencies](https://asdf-vm.com/guide/getting-started.html#plugin-dependencies). `asdf-nodejs` is used as the example with the dependencies listed, however I believe these deps are no longer correct given how the `asdf-nodejs` plugin has grown and changed since those docs were written.

**Comment by augustobmoura** [comment]
asdf-nodejs doesn't have any other required dependencies aside from the ones already required by asdf-core (git and curl), you think we still need to list those?

**Comment by jthegedus** [comment]
Right, well if the list https://asdf-vm.com/guide/getting-started.html#plugin-dependencies is no longer required we should probably remove them now. You could still use `help.deps` to codify and communicate the build deps for those moving target versions, but that's up to you.
